### DISCUSSION TITLE
"relative to" in a pause module is re-calculated over and over again

### DISCUSSION THREAD

**Discussion by mfellmann** [rubenarslan/formr.org#556] [Q&A]
Dear Community, 

we have a simple yet impactful bug in our formr instance that we could not resolve for years and tried everything. 
If we have a pause-module with a "relative to" and in this field, we simply put
`toString( now() + 60*5 )`
... then the expiry date is **initially** set as expected (i.e., current time plus 5 min.). 
However, **each time the user re-enters the run** e.g. by refreshing the website, then the expiry date is **re-calculated** and it will again be current time + 5 minutes. In addition, the problem occurs when transitioning from 23:59:59 to 0:00:00. 

Apart from that, all other timing features work correct and as expected. That is, if we set a fixed hour or date in the pause function, then everything works reliably. 

We would greatly appreciate any help. 

Kind regards, 
Michael

**Comment by rubenarslan** [comment]
Hi Michael, the up-to-date version of formr no longer recalculates expiry times. In older versions, the way to solve this is by making the expiry time relative to the time the user arrived at the pause, like so

```
library(lubridate)
as.POSIXct(last(survey_unit_sessions$created)) + hours(5)
```

But it's definitely time to upgrade to a newer version of formr.

**Reply by mfellmann** [reply]
Thank you very much. We will test it with the newest formr-release and hopefully it works.
